clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java

```diff
@@ -538,87 +679,93 @@ private void handleColumnCommand() {
-            .handle();
-      }
-      if (line.hasOption(GravitinoOptions.DATATYPE)) {
-        String datatype = line.getOptionValue(GravitinoOptions.DATATYPE);
-        newUpdateColumnDatatype(url, ignore, metalake, catalog, schema, table, column, datatype)
-            .handle();
-      }
-      if (line.hasOption(GravitinoOptions.POSITION)) {
-        String position = line.getOptionValue(GravitinoOptions.POSITION);
-        newUpdateColumnPosition(url, ignore, metalake, catalog, schema, table, column, position)
-            .handle();
-      }
-      if (line.hasOption(GravitinoOptions.NULL)) {
-        if (line.getOptionValue(GravitinoOptions.NULL).equals("true")) {
-          newUpdateColumnNullability(url, ignore, metalake, catalog, schema, table, column, true)
-              .handle();
-        } else if (line.getOptionValue(GravitinoOptions.NULL).equals("false")) {
-          newUpdateColumnNullability(url, ignore, metalake, catalog, schema, table, column, false)
+    switch (command) {
+      case CommandActions.LIST:
+        newListColumns(url, ignore, metalake, catalog, schema, table).handle();
+        break;
+
+      case CommandActions.CREATE:
+        {
+          String datatype = line.getOptionValue(GravitinoOptions.DATATYPE);
+          String comment = line.getOptionValue(GravitinoOptions.COMMENT);
+          String position = line.getOptionValue(GravitinoOptions.POSITION);
+          boolean nullable =
+              !line.hasOption(GravitinoOptions.NULL)
+                  || line.getOptionValue(GravitinoOptions.NULL).equals("true");
+          boolean autoIncrement =
+              line.hasOption(GravitinoOptions.AUTO)
+                  && line.getOptionValue(GravitinoOptions.AUTO).equals("true");
+          String defaultValue = line.getOptionValue(GravitinoOptions.DEFAULT);
+
+          newAddColumn(
+                  url,
+                  ignore,
+                  metalake,
+                  catalog,
+                  schema,
+                  table,
+                  column,
+                  datatype,
+                  comment,
+                  position,
+                  nullable,
+                  autoIncrement,
+                  defaultValue)
               .handle();
+          break;
         }
-      }
-      if (line.hasOption(GravitinoOptions.AUTO)) {
-        if (line.getOptionValue(GravitinoOptions.AUTO).equals("true")) {
-          newUpdateColumnAutoIncrement(url, ignore, metalake, catalog, schema, table, column, true)
-              .handle();
-        } else if (line.getOptionValue(GravitinoOptions.AUTO).equals("false")) {
-          newUpdateColumnAutoIncrement(url, ignore, metalake, catalog, schema, table, column, false)
-              .handle();
+
+      case CommandActions.DELETE:
+        newDeleteColumn(url, ignore, metalake, catalog, schema, table, column).handle();
+        break;
+
+      case CommandActions.UPDATE:
+        {
```

Comment:
line 723 and 764 can be safely dropped.

Reply:
No. they cannot, due to variable scope issues. Unfortunately, variables in a cases scope is the switch block not just the case itself.

Reply:
In this case, comment is defined in both the create and update cases, if they were to be removed the code will not compile.